.github/workflows/ci.yml

```diff
@@ -43,20 +43,35 @@ jobs:
 
   ormolu:
     runs-on: ubuntu-latest
+    env:
+      STACK_RESOLVER: lts-21.25
     steps:
       - uses: actions/checkout@v3
-      - uses: mrkkrp/ormolu-action@v11
+      - uses: extractions/setup-just@v2
+
+      - name: Install hpack
+        run: sudo apt install -y hpack
+
+      - name: Generate juvix.cabal
+        run: hpack
+
+      - name: Cache stack
+        id: stack-cache
+        uses: actions/cache@v3
         with:
-          version: 0.5.3.0
-          extra-args: >-
-            --ghc-opt -XDerivingStrategies
-            --ghc-opt -XImportQualifiedPost
-            --ghc-opt -XMultiParamTypeClasses
-            --ghc-opt -XPatternSynonyms
-            --ghc-opt -XStandaloneDeriving
-            --ghc-opt -XTemplateHaskell
-            --ghc-opt -XUnicodeSyntax
-            --ghc-opt -XBangPatterns
+          path: |
+            ~/.stack
```

Comment:
This works fine but did you consider the following?

1. Cache the path `~/.local/bin/ormolu` with key (e.g) `${{ runner.os }}-ormolu-${{ env.STACK_RESOLVER }}`
2. Guard the install ormolu step with `if: steps.stack-cache.outputs.cache-hit != 'true'` (`stack-cache` here refers to the `id` of the `Cache stack` step, see [example cache workflow](https://github.com/actions/cache?tab=readme-ov-file#example-cache-workflow))

This way the `stack install` step will not run if the `ormolu` binary already exists. This would avoid the hackage cache download triggered by `stack install` on each ormolu run.

Reply:
I've applied your suggestion. Thanks!